### PR TITLE
Add check finality RPC endpoints and updates consensus-finality.md

### PR DESCRIPTION
**Pull request by hyd628**
### Description

Adds the two new Moonbeam RPC endpoints  (https://github.com/PureStake/moonbeam/pull/1184) and sample curls.

Combines the finality checking scripts using various libraries from before into tabbed sections. 

### Checklist

- [x] If this requires translations for the `moonbeam-docs-cn` repo, I have created a ticket for the translations in Jira
- [x] If pages have been moved around, I have created an additional PR in `moonbeam-mkdocs` to update redirects
- [x] If images have been added, I have run the `compress-images.py` script to compress the images.
- [x] If images have been moved, I have created an additional PR in `moonbeam-docs-cn` to update images to their new paths
- [x] If variables (in variables.yml) need to be updated (such as a name change), I have updated the `moonbeam-docs-cn` repo to use the new variables

### Corresponding PRs

Please link to any corresponding PRs here.

### After Translation Requirements

- [ ] Will need to create PR in `moonbeam-docs` repo to remove images
- [ ] Will need to create PR in `moonbeam-docs` repo to remove variables
- [ ] Will need to create PR in `moonbeam-mkdocs` repo to add redirects for Chinese site
- [ ] No additional PRs are required after the translations are done

#### Items to be Updated

Please list any of the items that will need to be added or deleted after the translations are done here.

### CODE REVIEW

File: builders/get-started/eth-compare/consensus-finality.md [hunk 1]
```diff
@@ -42,61 +42,81 @@ However, the deterministic finality of Moonbeam can be used to provide a better
...
+
+Moonbeam has added support for two new RPC endpoints, `moon_isBlockFinalized` and `moon_isTxFinalized` to the Moonbeam node, that are useful for checking whether an on-chain event is finalized. The information on these two endpoints are as follows:
+
+=== "moon_isBlockFinalized"
+    |   Variable   |                                      Value                                       |
+    |:------------:|:--------------------------------------------------------------------------------|
+    |   Endpoint |                        `moon_isBlockFinalized`                     |
+    |   Description   | Check for the finality of the block given by its block hash. |
+    |  Parameters |    `block_hash`: **STRING** The hash of the block, accepts either Substrate-style or ethereum-style block hash as its input.                     | 
+    |  Returns | `result`: **BOOLEAN** Returns `True` if the block is finalized, `False` if the block is not finalized or not found.  |
```
Comment (eshaben) [inline builders/get-started/eth-compare/consensus-finality.md:55]: the parameters and returns rows look a little bit weird

also are `True` and `False` supposed to be capitalized?

and lastly, I would remove the period at the end of all of the sentences in tables

**Review by eshaben — CHANGES_REQUESTED** [review]
thanks for adding the header attributes! 😄 

can you please add {target=blank} to the links?

File: builders/get-started/eth-compare/consensus-finality.md (hunk 1 above)
Comment (hyd628) [reply builders/get-started/eth-compare/consensus-finality.md:55]: Removed the periods, and changed true and false to lower case to match with the results. Thanks.

**Comment by hyd628** [comment]
Added them. Thanks!

File: builders/build/moonbeam-custom-api.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+=== "moon_isBlockFinalized"
+    ```
+    curl -H "Content-Type: application/json" -X POST --data 
+        '[{
+            "jsonrpc":"2.0",
+            "id":"1",
+            "method":"moon_isBlockFinalized",
+            "params":["Put-Block-Hash-Here"
+        ]}]' 
+        https://rpc.api.moonbase.moonbeam.network
```
Comment (eshaben) [inline builders/build/moonbeam-custom-api.md:41]: these examples should use the `rpc_url` variable

File: builders/build/moonbeam-custom-api.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+=== "moon_isTxFinalized"
+    |   Variable   |                                      Value                                       |
+    |:------------:|:--------------------------------------------------------------------------------|
+    |   Endpoint |                        `moon_isTxFinalized`                     |
+    |   Description   | Check for the finality of the transaction given by its EVM tx hash |
+    |  Parameters | `tx_hash`: **STRING** The EVM tx hash of the transaction  | 
+    |  Returns |  `result`: **BOOLEAN** Returns `true` if the tx is finalized; `false` if the tx is not finalized or not found | 
+
+You can try out these endpoints with the following curl examples. These examples query the public RPC endpoint of Moonbase Alpha, but they can be modified to use with Moonbeam and Moonriver by changing the URL of the RPC endpoint to the corresponding [endpoints](https://docs.moonbeam.network/builders/get-started/endpoints/){target=blank}.
```
Comment (eshaben) [inline builders/build/moonbeam-custom-api.md:30]: I would reword this section a little bit, because they will need to sign up for their own API endpoint/api key... maybe something like...

`... but they can be modified to use with Moonbeam and Moonriver by changing the URL of the RPC endpoint to use your own endpoint and API key which you can get from one of the supported [Endpoint Providers](/builders/get-started/endpoints/){target=_blank}`

File: builders/get-started/eth-compare/consensus-finality.md [hunk 4]
```diff
@@ -42,63 +42,83 @@ However, the deterministic finality of Moonbeam can be used to provide a better
 
 The following sections outline how you can check for transaction finality using both the Ethereum JSON-RPC (custom Web3 request) and the Substrate (Polkadot) JSON-RPC.
 
-## Checking Tx Finality with Ethereum Libraries
-
-You can make calls to the Substrate JSON-RPC using the `send` method of both [Web3.js](https://web3js.readthedocs.io/) and [Ethers.js](https://docs.ethers.io/).
-
-Custom RPC requests are also possible using [Web3.py](https://web3py.readthedocs.io/) with the `make_request` method. You can use the Web3.js example as a baseline.
+## Checking Tx Finality with Moonbeam RPC Endpoints {: #checking-tx-finality-with-moonbeam-rpc-endpoints }
```
Comment (eshaben) [inline builders/get-started/eth-compare/consensus-finality.md:45]: I feel like this section shouldn't be duplicated... maybe just have the curl examples here and say `For more information you can go to the [Finality RPC Endpoints](/builders/build/moonbeam-custom-api#finality-rpc-endpoints){target=_blank} section of the Moonbeam Custom API page.` 

Not sure what the best path forward here is, if you have any ideas let me know! 🙂

**Review by eshaben — COMMENTED** [review]
Thanks for fixing a bunch of typos in this PR!!

I like the banner image but I feel like maybe the index page icon should just be the Moonbeam logo so it matches with the Ethereum & Substrate API pages

File: builders/get-started/eth-compare/consensus-finality.md (hunk 4 above)
Comment (hyd628) [reply builders/get-started/eth-compare/consensus-finality.md:45]: Good catch. I meant to revisit this part after splitting out the page.

**Comment by hyd628** [comment]
Thanks Erin, I've incorporated the feedback and swapped the icon for Moonbeam logo.

**Review by eshaben — APPROVED** [review]
Sweeeet! Thanks!

LGTM ✨
